extensions/src/ACRE2Arma/arma2ts/CallExt_DllMain.cpp

```diff
@@ -192,7 +192,7 @@ void __stdcall RVExtension(char *output, int outputSize, const char *function)
                 // allows up to 4096 char string (+1 to ensure NUL terminated), like initial version
 
                 //DEBUG("Read from pipe [%d]\n", hPipe);
-                const BOOL ret = ReadFile(readHandle, (LPVOID)value, sizeof(char) * (read_length - 1U), &cbRead, NULL);
+                const bool ret = ReadFile(readHandle, (LPVOID)value, sizeof(char) * (read_length - 1U), &cbRead, NULL);
```

Comment:
This is unfortunately not related to C++ standard, but in correctly using the Windows specific functions:
https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-readfile. As an example, this one uses BOOL instead of bool for a return type.

Reply:
As long as it isn't redefined I think you can repeat the typedef multiple times: e.g.:
```
typedef int BOOL;
typedef int BOOL;
```
so we could just add `typedef int BOOL;` at the top, which should make it portable without having to change type

Reply:
@PabstMirror you're right, I think it could be moved to a file called `compat_linux.h` or something, to be decided :grinning: 

@TheMagnetar fixed

Reply:
What is fixed? I mean, why change `BOOL` to  `bool`? There is  no gain since the majority of that code is Windows specific and the API states BOOL instead of bool. Or am I seeing some unpublished code :)

The same applies in my opinion to a `compat_linux.h` file. The only solution I saw that was viable without a bunch of compats was using SDL2 for having the same code on windows and on linux. However, not everything can be done in that way (loading PBOs for example)

Reply:
> There is no gain since the majority of that code is Windows specific and the API states BOOL instead of bool. Or am I seeing some unpublished code :)

Currently no gain from this, you're right. I changed the commit to replace some of TRUE/FALSE directives, doesn't change behavior, but for cross-platform code that's one less constant to have.